### PR TITLE
Add an option to use esp-alloc as the global allocator

### LINKED ISSUE #24: Add an option for an alloc crate
I'm new to rust on embedded systems. I want to use esp-hal instead of esp-idf because idf has lots of stuff and configuration I dont understand, so being more bare metal will help me learn here.

I am attempting to use the alloc crate, it would be great if the cargo generate for this template prompted the user to ask them if they want to use `alloc`, and gives them a good default global allocator to use, like `esp-alloc`

### PR DESCRIPTION
**Pull request by jessebraham**
Closes #24

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -12,8 +16,39 @@ use xtensa_atomic_emulation_trap as _;
 use xtensa_lx_rt::entry;
 {%- endif %}
 
+{% if alloc -%}
+#[global_allocator]
+static ALLOCATOR: esp_alloc::EspHeap = esp_alloc::EspHeap::empty();
+
+fn init_heap() {
+    const HEAP_SIZE: usize = 32 * 1024;
```
Comment (jessebraham) [inline src/main.rs:15]: Is this a reasonable default value?

**Comment by jessebraham** [comment]
I don't have the energy to fight with CI right now so I'll finish this some other time

File: src/main.rs (hunk 1 above)
Comment (bjoernQ) [reply src/main.rs:15]: I think we can start with this and maybe add a comment for the user

The best solution would be to add the heap_end symbol for ESP32-C3 and use the values provided by the linker script, certainly

**Comment by SergioGasquez** [comment]
Hello! [Here is how the CI could be fixed](https://github.com/esp-rs/esp-template/compare/main...SergioGasquez:esp-template:feature/alloc#diff-e667cbf8e98c5f2339128886eb5203cc3e3cfc6452374cdf97af98c92b2485ef)! Feel free to copy the changes and update this PR or let me know if you want me to open a new PR and I'll happily do it.

**Pull request closed without merging**